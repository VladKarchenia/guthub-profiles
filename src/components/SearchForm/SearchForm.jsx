// import React, { useEffect, useContext, Fragment } from "react";
import React, { useEffect, Fragment } from "react";
import { SearchResults } from "../SearchResults/SearchResults.jsx";
// import { SearchResultsContext } from "../../context/searchResultsContext.js";
import { Loader } from "../Loader/Loader.jsx";
// import styles from "./SearchForm.module.scss";
// import Button from "@material-ui/core/Button";
// import Icon from "@material-ui/core/Icon";
import TextField from "@material-ui/core/TextField";
// import { styled } from "@material-ui/core/styles";
import { withStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";
import { setSearchValue, thunkFetchUsers } from "../../actions/actions.js";
import { useHistory } from "react-router-dom";
import PropTypes from "prop-types";

// const StyledButton = styled(Button)({
//   background: "#e21e35",
//   width: "15%",
//   height: 54,
//   minWidth: 110,
//   borderRadius: "0 5px 5px 0",
//   fontSize: 15,
//   fontWeight: 600,
//   letterSpacing: "0.005em",
//   padding: "2px 0 0",
//   boxShadow: "0 2px 1px #c20017",
//   "&:hover, &:active": {
//     background: "#bf0016",
//     boxShadow: "0 2px 1px #9e0012"
//   },
//   "& .MuiButton-label": {
//     fontWeight: 600
//   }
// });

const StyledInput = withStyles({
  root: {
    margin: "50px 0 30px",
    fontSize: 17,
    color: "#333",
    fontWeight: 600,
    borderRadius: 5,
    "& .MuiOutlinedInput-root": {
      borderRadius: 5,
    }
  }
})(TextField);

const SearchForm = ({ searchValue, setSearchValue, thunkFetchUsers, isLoading, users }) => {
  // const { isLoading, users, findUsers } = useContext(SearchResultsContext);
  let history = useHistory();

  useEffect(() => {
    let searchValueFromHistory = history.location.search.replace("?search=", "");
    if (searchValueFromHistory) {
      thunkFetchUsers(searchValueFromHistory);
    }
  }, []);

  // const handleSubmit = (e) => {
  //   e.preventDefault();
  //   if (searchValue.trim()) {
  //     thunkFetchUsers(searchValue);
  //     setSearchValue("");
  //   }
  // };

  const handleSearchChange = (e) => {
    const newValue = e.target.value;
    setSearchValue(newValue);
    history.push({
      pathname: "/",
      search: `?search=${newValue}`,
    });
  };

  return (
    <Fragment>
      {/* <form className={styles.search} onSubmit={handleSubmit} noValidate autoComplete='off'>
        <StyledInput id='outlined-search' label='Find a user' type='search' variant='outlined' value={searchValue} onChange={handleSearchChange} fullWidth />
        <StyledButton variant='contained' color='secondary' type='submit'>
          <Icon>search</Icon>
          Search
        </StyledButton>
      </form> */}
      <StyledInput id='outlined-search' label='Find a user' type='search' variant='outlined' value={searchValue} onChange={handleSearchChange} fullWidth autoComplete='off' />
      {isLoading ?
        <Loader />
        :
        <SearchResults
          users={
            users && users.length > 0
              ? users
              : []
          }
        />
      }
    </Fragment>
  );
};

export default connect(
  (state) => ({
    searchValue: state.findUsers.searchValue,
    isLoading: state.findUsers.isLoading,
    users: state.findUsers.users,
  }),
  {
    setSearchValue,
    thunkFetchUsers,
  }
)(SearchForm);

SearchForm.propTypes = {
  searchValue: PropTypes.string,
  setSearchValue: PropTypes.func,
  isLoading: PropTypes.bool,
  users: PropTypes.array,
  thunkFetchUsers: PropTypes.func,
};
