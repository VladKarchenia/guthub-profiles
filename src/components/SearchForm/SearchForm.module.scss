.search {
  display: flex;
  width: 100%;
  margin: 50px 0 30px;
}
