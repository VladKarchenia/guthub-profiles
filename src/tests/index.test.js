import React from "react";
import { Provider } from "react-redux";
import SearchForm from "../components/SearchForm/SearchForm";
import renderer from "react-test-renderer";
import configureStore from "../index";

describe("Search form component", () => {
  it("matches the snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={configureStore()}>
          <SearchForm />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
