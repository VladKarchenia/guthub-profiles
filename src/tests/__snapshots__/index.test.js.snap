// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search form component matches the snapshot 1`] = `
Array [
  <div
    className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-3 MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-search"
      id="outlined-search-label"
    >
      Find a user
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="search"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-6"
        >
          <span>
            Find a user
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
  <div
    className="found_number"
  >
    You can find any GitHub user you want
  </div>,
  <div
    className="results"
  />,
]
`;
